src/fund/components/FundCardAmountInputTypeSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useMemo } from 'react';
+import { useIcon } from '../../core-react/internal/hooks/useIcon';
+import { getRoundedAmount } from '../../core/utils/getRoundedAmount';
+import { Skeleton } from '../../internal/components/Skeleton';
+import { cn, color, pressable, text } from '../../styles/theme';
+import type { FundCardAmountInputTypeSwitchPropsReact } from '../types';
+
+export const FundCardAmountInputTypeSwitch = ({
+  selectedInputType,
+  setSelectedInputType,
+  selectedAsset,
+  fundAmountFiat,
+  fundAmountCrypto,
+  exchangeRate,
+  isLoading,
+}: FundCardAmountInputTypeSwitchPropsReact) => {
+  const iconSvg = useIcon({ icon: 'toggle' });
+
+  const handleToggle = () => {
+    setSelectedInputType(selectedInputType === 'fiat' ? 'crypto' : 'fiat');
+  };
+
+  const formatUSD = useCallback((amount: string) => {
+    if (!amount || amount === '0') {
+      return null;
+    }
+    const roundedAmount = Number(getRoundedAmount(amount, 2));
+    return `$${roundedAmount.toFixed(2)}`;
+  }, []);
+
+  const exchangeRateLine = useMemo(() => {
+    return (
+      <span
+        className={cn(
+          text.label2,
+          color.foregroundMuted,
+          'font-normal',
+          'pl-1',
+        )}
+      >
+        ({formatUSD('1')} = {exchangeRate?.toFixed(8)} {selectedAsset})
+      </span>
+    );
+  }, [formatUSD, exchangeRate, selectedAsset]);
+
+  const cryptoAmountLine = useMemo(() => {
+    return (
+      <span className={cn(text.label1)}>
+        {Number(fundAmountCrypto).toFixed(8)} {selectedAsset}
+      </span>
+    );
+  }, [fundAmountCrypto, selectedAsset]);
+
+  const fiatAmountLine = useMemo(() => {
+    return <span className={cn(text.label1)}>{formatUSD(fundAmountFiat)}</span>;
+  }, [formatUSD, fundAmountFiat]);
```

Comment:
nit: this would be better as a single useMemo that returned the correctAmount line based on type